### ISSUE TITLE
eth.getTransaction() throws error

### ISSUE THREAD

**Issue by xuhcc** [pipermerriam/web3.py#273]
* Version: 3.13.3
* Python: 3.6.2
* OS: Arch Linux

### What was wrong?

I'm trying to get transaction by hash on Rinkeby testnet from Geth v1.6.7 node:

```python
>>> tx_hash = '0xe5854bda43d2dff2c4c809e044ad5ac9c1f34ff22dbf0931459858575896c967'
>>> web3.eth.getTransaction(tx_hash)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/venv/lib/python3.6/site-packages/web3/eth.py", line 160, in getTransaction
    [transaction_hash],
  File "/venv/lib/python3.6/site-packages/web3/manager.py", line 112, in request_blocking
    response = self._make_request(method, params)
  File "/venv/lib/python3.6/site-packages/web3/manager.py", line 95, in _make_request
    return make_request_fn(method, params)
  File "/venv/lib/python3.6/site-packages/web3/middleware/attrdict.py", line 20, in middleware
    response = make_request(method, params)
  File "/venv/lib/python3.6/site-packages/web3/middleware/formatting.py", line 32, in middleware
    formatter(response['result']),
  File "cytoolz/functoolz.pyx", line 236, in cytoolz.functoolz.curry.__call__ (cytoolz/functoolz.c:4144)
  File "cytoolz/functoolz.pyx", line 232, in cytoolz.functoolz.curry.__call__ (cytoolz/functoolz.c:3996)
  File "/venv/lib/python3.6/site-packages/web3/utils/formatters.py", line 45, in apply_formatter_if
    return formatter(value)
  File "cytoolz/functoolz.pyx", line 236, in cytoolz.functoolz.curry.__call__ (cytoolz/functoolz.c:4144)
  File "cytoolz/functoolz.pyx", line 232, in cytoolz.functoolz.curry.__call__ (cytoolz/functoolz.c:3996)
  File "/venv/lib/python3.6/site-packages/eth_utils/functional.py", line 33, in inner
    return callback(fn(*args, **kwargs))
  File "/venv/lib/python3.6/site-packages/web3/utils/formatters.py", line 55, in apply_formatters_to_dict
    yield key, formatters[key](item)
  File "cytoolz/functoolz.pyx", line 236, in cytoolz.functoolz.curry.__call__ (cytoolz/functoolz.c:4144)
  File "cytoolz/functoolz.pyx", line 232, in cytoolz.functoolz.curry.__call__ (cytoolz/functoolz.c:3996)
  File "/venv/lib/python3.6/site-packages/web3/utils/formatters.py", line 44, in apply_formatter_if
    if condition(value):
  File "cytoolz/functoolz.pyx", line 608, in cytoolz.functoolz.complement.__call__ (cytoolz/functoolz.c:10825)
  File "/venv/lib/python3.6/site-packages/web3/middleware/pythonic.py", line 64, in is_create_address
    return decode_hex(value) == b""
  File "/venv/lib/python3.6/site-packages/eth_utils/string.py", line 78, in inner
    return force_obj_to_bytes(fn(*args, **kwargs))
  File "/venv/lib/python3.6/site-packages/eth_utils/hexidecimal.py", line 22, in decode_hex
    raise TypeError('Value must be an instance of str or unicode')
TypeError: Value must be an instance of str or unicode
```

In Geth console:

```js
> eth.getTransaction('0xe5854bda43d2dff2c4c809e044ad5ac9c1f34ff22dbf0931459858575896c967')
{
  blockHash: "0xea841221179e37ca9cc23424b64201d8805df327c3296a513e9f1fe6faa5ffb3",
  blockNumber: 3911,
  from: "0xbedcf417ff2752d996d2ade98b97a6f0bef4beb9",
  gas: 4700000,
  gasPrice: 20000000000,
  hash: "0xe5854bda43d2dff2c4c809e044ad5ac9c1f34ff22dbf0931459858575896c967",
  input: "0x6060604052341561000c57fe5b5b6101828061001c6000396000f30060606040526000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff168063230925601461003b575bfe5b341561004357fe5b61008360048080356000191690602001909190803560ff1690602001909190803560001916906020019091908035600019169060200190919050506100c5565b604051808273ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200191505060405180910390f35b6000600185858585604051806000526020016040526000604051602001526040518085600019166000191681526020018460ff1660ff1681526020018360001916600019168152602001826000191660001916815260200194505050505060206040516020810390808403906000866161da5a03f1151561014257fe5b50506020604051035190505b9493505050505600a165627a7a7230582054abc8e7b2d8ea0972823aa9f0df23ecb80ca0b58be9f31b7348d411aaf585be0029",
  nonce: 0,
  r: "0xe675b8854c227bb7bd0c5db4b64b1846447a972e1f87e7420f8644dacc3f36ec",
  s: "0x69c5abc563cb59f6b770dbd52a72f0ce24640067b66cbcecd9ae161a031c1ec8",
  to: null,
  transactionIndex: 0,
  v: "0x2b",
  value: 0
}
```

**Comment by carver** [comment]
The key feature of this transaction appears to be that it's a contract creation (ie~ `to` is null). I am looking for a test case for that right now, and will add one if I don't find it.

**Comment by pipermerriam** [comment]
@xuhcc this is fixed and available in `3.13.4`

**Comment by xuhcc** [comment]
@pipermerriam 

Thanks!
